bin/tsv_for_genome_upload.py

```diff
@@ -0,0 +1,386 @@
+            paths.extend([os.path.join(self.absolute_path, PROK_PATH, os.path.basename(i)) for i in self.prok_mag])
+        return genomes, stats_software, paths
+
+    def get_assembly_software(self, genomes):
+        assembly_software = {}
+        software_list = []
+        with open(self.assembly_software_file, 'r') as file_in:
+            for line in file_in:
+                line = line.strip().split('\t')
+                assembly_software[line[0]] = line[1]
+        for genome in genomes:
+            name = genome.split('_')[0]
+            software_list.append(assembly_software[name])
+        return software_list
+
+    def get_stats(self, input_file):
+        stats = {}
+        if not input_file:
+            return stats
+        with open(input_file, 'r') as file_in:
+            for line in file_in:
+                if "completeness" in line:
+                    continue
+                line = line.strip().split(',')
+                comp = round(float(line[1]), 2)
+                if comp == 100.0:
+                    comp = 100
+                cont = round(float(line[2]), 2)
+                stats[line[0]] = [comp, cont]
+        return stats
+
+    def get_coverage(self, files):
+        coverage = {}
+        if not files:
+            return coverage
+        for coverage_file in files:
+            with open(coverage_file, 'r') as file_in:
+                for line in file_in:
+                    line = line.strip().split('\t')
+                    coverage[line[0]] = round(float(line[1]), 2)
+        return coverage
+
+    def get_rna(self, genomes):
+        rrna, trna = {}, {}
+        if not self.rna:
+            return ['False' for _ in genomes]
+        for filename in self.rna:
+            if filename.endswith('_rRNAs.out'):
+                genome = filename.split('_rRNAs.out')[0] + '.fa'
+                rrna[genome] = self.check_rna(filename, LIMIT_RRNA)
+            if filename.endswith('_tRNA_20aa.out'):
+                genome = filename.split('_tRNA_20aa.out')[0] + '.fa'
+                trna[genome] = self.check_rna(filename, LIMIT_TRNA)
+        final_decision = []
+        for genome in genomes:
+            if genome not in trna or genome not in rrna:
+                final_decision.append('False')
+            else:
+                if rrna[genome] and trna[genome]:
+                    final_decision.append('True')
+                else:
+                    final_decision.append('False')
+        return final_decision
+
+    def check_rna(self, filename, limit):
+        with open(filename, 'r') as file_in:
+            for line in file_in:
+                cur_rna_count = float(line.strip().split('\t')[1])
+                if cur_rna_count < limit:
+                    return False
+        return True
+
+    def get_taxonomy(self, genomes):
+        final_tax = []
+        lineage = {}
+        if self.tax_euks:
```

Comment:
Could this become
```
for tax in (self.tax_euks, self.tax_proks):
     if tax:
          with open(tax, 'r') as file_in:
               for line in file_in:
                    if 'classification' in line:
                         continue
                    else:
                         line = line.strip().split('\t')
                         if tax == self.tax_euks:
                              lineage[line[0]] = line[3]
                         else:
                              lineage[line[0]+'.fa'] = line[2]
```
?

Reply:
And by the way I approve of keeping the coassembly methods here, even if disabled. They are potential material for the future.

Reply:
improved :)